keras/utils/audio_dataset.py

```diff
@@ -0,0 +1,276 @@
+      seed: Optional random seed for shuffling and transformations.
+      validation_split: Optional float between 0 and 1,
+          fraction of data to reserve for validation.
+      subset: Subset of the data to return.
+          One of "training" or "validation".
+          Only used if `validation_split` is set.
+      follow_links: Whether to visits subdirectories pointed to by symlinks.
+          Defaults to False.
+
+    Returns:
+      A `tf.data.Dataset` object.
+        - If `label_mode` is None, it yields `string` tensors of shape
+          `(batch_size,)`, containing the contents of a batch of audio files.
+        - Otherwise, it yields a tuple `(audio, labels)`, where `audio`
+          has shape `(batch_size, sequence_length, num_channels)` and `labels` follows the format described
+          below.
+
+    Rules regarding labels format:
+      - if `label_mode` is `int`, the labels are an `int32` tensor of shape
+        `(batch_size,)`.
+      - if `label_mode` is `binary`, the labels are a `float32` tensor of
+        1s and 0s of shape `(batch_size, 1)`.
+      - if `label_mode` is `categorical`, the labels are a `float32` tensor
+        of shape `(batch_size, num_classes)`, representing a one-hot
+        encoding of the class index.
+    """
+    if labels not in ("inferred", None):
+        if not isinstance(labels, (list, tuple)):
+            raise ValueError(
+                "`labels` argument should be a list/tuple of integer labels, of "
+                "the same size as the number of audio files in the target "
+                "directory. If you wish to infer the labels from the subdirectory "
+                'names in the target directory, pass `labels="inferred"`. '
+                "If you wish to get a dataset that only contains audio samples "
+                f"(no labels), pass `labels=None`. Received: labels={labels}"
+            )
+        if class_names:
+            raise ValueError(
+                "You can only pass `class_names` if "
+                f'`labels="inferred"`. Received: labels={labels}, and '
+                f"class_names={class_names}"
+            )
+    if label_mode not in {"int", "categorical", "binary", None}:
+        raise ValueError(
+            f'`label_mode` argument must be one of "int", "categorical", "binary", or None. Received: label_mode={label_mode}'
+        )
+
+    if not ragged and output_sequence_length is None:
+        raise ValueError(
+            f"The dataset should be ragged dataset or fixed sequence length dataset, found ragged={ragged} and output_sequence_length={output_sequence_length}"
+        )
+    elif ragged and output_sequence_length is not None:
+        raise ValueError("Cannot set both `ragged` and `output_sequence_length`")
+
+    if sampling_rate is not None:
+        if not isinstance(sampling_rate, int):
+            raise ValueError("`sampling_rate` should have an integer value")
```

Comment:
The error message should display what was passed by the user. `Received: sampling_rate={sampling_rate}`